### PR TITLE
[CM-699] Show rating same as web on agent app - iOS

### PR DESCRIPTION
**Pull request by Sathyan-Elangovan**
## Summary
- Show rating same as web.
in iOS sdk, we are just showing `you rated the conversation` message on the conversation after the feedback.But the improvement on this is we need to show rated emoji as well as comments(if any)

Current behaviour:

Improvement:

### CODE REVIEW

File: Sources/Views/ALKInformationCell.swift [hunk 1]
```diff
@@ -55,7 +57,13 @@ final class ALKInformationCell: UITableViewCell {
                                                                    options: NSStringDrawingOptions.usesLineFragmentOrigin,
                                                                    attributes: [NSAttributedString.Key.font: ALKMessageStyle.infoMessage.font],
                                                                    context: nil)
-            messageHeigh = rect.height + 17
+            //get feedback dictionary for view
+            guard let dictionary = ALKInformationCell().getFeedback(viewModel: viewModel) else { return 0 }
+            if dictionary["comments"] != nil {
+                messageHeigh = (rect.height + 17 + 30)
```
Comment (mukeshthawani) [inline Sources/Views/ALKInformationCell.swift:63]: Create an enum for paddings.

File: Sources/Views/ALKInformationCell.swift [hunk 2]
```diff
@@ -76,31 +84,110 @@ final class ALKInformationCell: UITableViewCell {
...
+        textString.append(imageString)
+        messageView.attributedText = textString
+        if !comment.isEmpty {
+            commentTextView.text = "“\(comment)”"
+        }else{
+            commentTextView.text = ""
+        }
+        
+        setUpConstraintsForRating()
+        setupStyle()
```
Comment (mukeshthawani) [inline Sources/Views/ALKInformationCell.swift:122]: This we may have to call even the feedback is not present.

File: Sources/Views/ALKInformationCell.swift [hunk 3]
```diff
@@ -76,31 +84,110 @@ final class ALKInformationCell: UITableViewCell {
 
     func update(viewModel: ALKMessageViewModel) {
         self.viewModel = viewModel
+        
+        guard let dictionary = getFeedback(viewModel: viewModel) else { return }
```
Comment (mukeshthawani) [inline Sources/Views/ALKInformationCell.swift:88]: Name it as feedback or something.

File: Sources/Views/ALKInformationCell.swift [hunk 4]
```diff
@@ -76,31 +84,110 @@ final class ALKInformationCell: UITableViewCell {
...
+        lineViewLeft.backgroundColor = .lightGray
+        lineViewRight.backgroundColor = .lightGray
+        
+        contentView.addViewsForAutolayout(views: [horizontalStackView])
+        contentView.bringSubviewToFront(messageView)
+        
+        horizontalStackView.leadingAnchor.constraint(equalTo: contentView.leadingAnchor, constant: 10).isActive = true
+        horizontalStackView.trailingAnchor.constraint(equalTo: contentView.trailingAnchor, constant: -10).isActive = true
+        
+        lineViewLeft.widthAnchor.constraint(equalToConstant: 60).isActive = true
```
Comment (mukeshthawani) [inline Sources/Views/ALKInformationCell.swift:172]: Same here, move all the values to a separate enum.

File: Sources/Views/ALKInformationCell.swift [hunk 5]
```diff
@@ -76,31 +84,110 @@ final class ALKInformationCell: UITableViewCell {
...
+        
         messageView.setFont(ALKMessageStyle.infoMessage.font)
         messageView.textColor = ALKMessageStyle.infoMessage.text
+        commentTextView.setFont(ALKMessageStyle.feedbackComment.font)
+        commentTextView.textColor = ALKMessageStyle.feedbackComment.text
+    }
+    
+    func getFeedback(viewModel: ALKMessageViewModel) -> Dictionary<String,Any>? {
+        guard let feedbackString = viewModel.metadata?["feedback"] as? String else { return nil }
+        guard let dictionary = try? JSONSerialization.jsonObject(with: feedbackString.data(using: .utf8)!, options: []) as? [String:Any] else { return nil }
```
Comment (mukeshthawani) [inline Sources/Views/ALKInformationCell.swift:190]: Don't force unwrap the feedback string data.

**Review by mukeshthawani — COMMENTED** [review]
Add a Changelog if we'll show this to the user as well.

File: Sources/Views/ALKInformationCell.swift (hunk 1 above)
Comment (Sathyan-Elangovan) [reply Sources/Views/ALKInformationCell.swift]: done

File: Sources/Views/ALKInformationCell.swift (hunk 2 above)
Comment (Sathyan-Elangovan) [reply Sources/Views/ALKInformationCell.swift:163]: done

File: Sources/Views/ALKInformationCell.swift (hunk 3 above)
Comment (Sathyan-Elangovan) [reply Sources/Views/ALKInformationCell.swift]: done

File: Sources/Views/ALKInformationCell.swift (hunk 4 above)
Comment (Sathyan-Elangovan) [reply Sources/Views/ALKInformationCell.swift]: added enum for padding

File: Sources/Views/ALKInformationCell.swift (hunk 5 above)
Comment (Sathyan-Elangovan) [reply Sources/Views/ALKInformationCell.swift]: done

**Review by mukeshthawani — APPROVED** [review]

**Pull request merged**